godot-core/src/task/futures.rs

```diff
@@ -62,42 +74,51 @@ impl<T> Default for SignalFutureData<T> {
 }
 
 // Only public for itest.
-#[cfg_attr(feature = "trace", derive(Default))]
-pub struct SignalFutureResolver<R> {
-    data: Arc<Mutex<SignalFutureData<R>>>,
+pub struct SignalFutureResolver<R: IntoMaybeSend> {
+    data: Arc<Mutex<SignalFutureData<R::Target>>>,
 }
 
-impl<R> Clone for SignalFutureResolver<R> {
+impl<R: IntoMaybeSend> Clone for SignalFutureResolver<R> {
     fn clone(&self) -> Self {
         Self {
             data: self.data.clone(),
         }
     }
 }
 
-impl<R> SignalFutureResolver<R> {
-    fn new(data: Arc<Mutex<SignalFutureData<R>>>) -> Self {
+// Implement Default for itest if R is also Default.
+#[cfg(feature = "trace")]
+impl<R: IntoMaybeSend + Default> Default for SignalFutureResolver<R> {
+    fn default() -> Self {
+        Self {
+            data: Default::default(),
+        }
+    }
+}
```

Comment:
If it's only needed by the test, we may as well have a utility function there.

`#[cfg(feature = "trace")]` is typically used to add tracing behavior for itests, not extra APIs (with maybe the exception of making a few symbols public, such as `CowArg`).

Reply:
> If it's only needed by the test, we may as well have a utility function there.

👍 

> #[cfg(feature = "trace")] is typically used to add tracing behavior for itests, not extra APIs (with maybe the exception of making a few symbols public, such as CowArg).

Should we introduce a new feature then?

Reply:
No, please not. We already have too many internal features.

Reply:
Maybe we just rename the feature to `itest` to broaden the scope? We can also do this later.

Reply:
I moved this into a function.